src/App.tsx

```diff
@@ -1,27 +1,63 @@
-import React from 'react';
+import React, { useCallback, useState } from 'react';
 import './App.scss';
 import { GoodsList } from './GoodsList';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+import { getAll, get5First, getRedGoods } from './api/goods';
+import { Good } from './types/Good';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+export const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[] | []>([]);
+  const [hasError, setHasError] = useState(false);
 
-    <button type="button" data-cy="all-button">
-      Load all goods
-    </button>
+  // eslint-disable-next-line @typescript-eslint/ban-types
+  const handleButton = useCallback((f: Function) => {
+    return async () => {
+      setHasError(false);
 
-    <button type="button" data-cy="first-five-button">
-      Load 5 first goods
-    </button>
+      try {
+        const selectedGoods = await f();
 
-    <button type="button" data-cy="red-button">
-      Load red goods
-    </button>
+        setGoods(selectedGoods);
+      } catch {
+        setHasError(true);
```

Comment:
it will be great to understand what kind of error we have, you can log this error in console
```suggestion
      } catch (error) {
        setHasError(true);
```